staging/src/k8s.io/apiserver/pkg/util/proxy/proxy.go

```diff
@@ -64,28 +65,28 @@ func ResolveEndpoint(services listersv1.ServiceLister, endpoints listersv1.Endpo
 	if len(eps.Subsets) == 0 {
 		return nil, errors.NewServiceUnavailable(fmt.Sprintf("no endpoints available for service %q", svc.Name))
 	}
-
-	// Pick a random Subset to start searching from.
-	ssSeed := rand.Intn(len(eps.Subsets))
-
-	// Find a Subset that has the port.
-	for ssi := 0; ssi < len(eps.Subsets); ssi++ {
-		ss := &eps.Subsets[(ssSeed+ssi)%len(eps.Subsets)]
-		if len(ss.Addresses) == 0 {
+	// Looking for a random IP address that matches the Service port:
+	// Go natively shuffles slice items when iterating over it.
+	for _, ep := range eps.Subsets {
+		if len(ep.Addresses) == 0 {
 			continue
 		}
-		for i := range ss.Ports {
-			if ss.Ports[i].Name == svcPort.Name {
-				// Pick a random address.
-				ip := ss.Addresses[rand.Intn(len(ss.Addresses))].IP
-				port := int(ss.Ports[i].Port)
+
+		for _, p := range ep.Ports {
+			if p.Name == svcPort.Name {
+				// Creating a new random generator to ensure non reproducibility
+				r := rand.New(rand.NewSource(time.Now().UnixNano()))
```

Comment:
a new random generator per request is unexpected

math.rand is seeded at process start in staging/src/k8s.io/component-base/cli/run.go

Reply:
The proposed change has been removed.